### PR TITLE
New formula: muttvcardsearch 1.9

### PR DESCRIPTION
**Pull request by benjaminfrank**
A small mutt carddav search utility for your Owncloud or SOGo server.

### CODE REVIEW

File: Library/Formula/muttvcardsearch.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Muttvcardsearch < Formula
+  desc "A small mutt carddav search utility for Owncloud or SOGo server"
+  homepage "https://github.com/BlackIkeEagle/muttvcardsearch"
+  url "https://github.com/BlackIkeEagle/muttvcardsearch/archive/v1.9.tar.gz"
+  sha256 "4c099938dd02f577d8289bbc5e82f0af6f290a564a30f6fdcb5cc880cd02fc8d"
+
+  option "with-debug", "build debug version"
+  depends_on "cmake" => :build
```
Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: Spacing between options and deps please.

File: Library/Formula/muttvcardsearch.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on "sqlite"
```
Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: Is the system version not recent enough?

File: Library/Formula/muttvcardsearch.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  def install
+    mkdir "build" do
+      args = std_cmake_parameters.split
+      if build.include? "enable-debug"
```
Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: `if build.with?`

File: Library/Formula/muttvcardsearch.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        args.delete "-DCMAKE_BUILD_TYPE=None"
```
Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: We don't pass this argument any more in `std_cmake_args`.

Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: We use `RELEASE` now.

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: Library/Formula/muttvcardsearch.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+class Muttvcardsearch < Formula
+  desc "A small mutt carddav search utility for Owncloud or SOGo server"
+  homepage "https://github.com/BlackIkeEagle/muttvcardsearch"
+  url "https://github.com/BlackIkeEagle/muttvcardsearch/archive/v1.9.tar.gz"
+  sha256 "4c099938dd02f577d8289bbc5e82f0af6f290a564a30f6fdcb5cc880cd02fc8d"
+
+  option "with-debug", "build debug version"
```
Comment (mikemcquaid) [inline Library/Formula/muttvcardsearch.rb]: What's the advantage of this? We don't typical add debug options.

File: Library/Formula/muttvcardsearch.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  def install
+    mkdir "build" do
+      args = std_cmake_parameters.split
+      if build.with? "debug"
+        args.delete "-DCMAKE_BUILD_TYPE=Release"
+        args << "-DCMAKE_BUILD_TYPE=Debug"
+      end
+      system "cmake", "..", *args
+      system "make"
+      bin.install "muttvcardsearch"
```
Comment (mikemcquaid) [inline Library/Formula/muttvcardsearch.rb]: Is there no `make install` we can use?

File: Library/Formula/muttvcardsearch.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    end
+  end
+
+  test do
+    # muttvcardsearch requires a valid carddav endpoint to
+    # successfully run any operation (rc == 0)
+    true
```
Comment (mikemcquaid) [inline Library/Formula/muttvcardsearch.rb]: Even e.g. `--version`? Really would like to have something here.

File: Library/Formula/muttvcardsearch.rb (hunk 5 above)
Comment (benjaminfrank) [inline Library/Formula/muttvcardsearch.rb]: Not that I am aware of, I just carried it along, to be totally honest. I'll remove it.

File: Library/Formula/muttvcardsearch.rb (hunk 7 above)
Comment (benjaminfrank) [inline Library/Formula/muttvcardsearch.rb]: the non-zero rc was the issue here, but today I learned that there is eg assert_match to check the output of a command, so the rc is not really releveant. Sorry, I am rather new to this whole formula thing. I'll change it to something meaningfull later.

Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: We can also "expect" the exit code if necessary. [Example](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/sonar.rb#L56-L58).

File: Library/Formula/muttvcardsearch.rb (hunk 6 above)
Comment (benjaminfrank) [inline Library/Formula/muttvcardsearch.rb]: I am no cmake expert, but using 'make install' aftwards gives the following error:
==> make install
  "/tmp/muttvcardsearch20150702-4863-so96fg/muttvcardsearch-1.9/build/muttvcardsearch"
  to "/usr/bin/muttvcardsearch".
make: *** [install] Error 1

File: Library/Formula/muttvcardsearch.rb (hunk 7 above)
Comment (benjaminfrank) [inline Library/Formula/muttvcardsearch.rb]: Nice, thanks.

File: Library/Formula/muttvcardsearch.rb (hunk 6 above)
Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: Rather than `std_cmake_parameters.split` you just want `std_cmake_args`, I believe.

**Comment by DomT4** [comment]
```
        ==> brew style muttvcardsearch
== /usr/local/Library/Formula/muttvcardsearch.rb ==
C: 19:  5: Incorrect indentation detected (column 4 instead of 5).
C: 20:  3: Use 2 (not 3) spaces for indentation.

1 file inspected, 2 offenses detected
```

**Comment by benjaminfrank** [comment]
Thanks, I couldn't view the Details yesterday.

**Comment by DomT4** [comment]
Yeah, we had some CI issues that persisted throughout yesterday. Sorry about the delay. You can hit the same audit input as the bot locally by doing `brew audit --strict` on the formula.

**Comment by benjaminfrank** [comment]
I thought I did, that's why I was puzzled there is an error...(aka lazy me)

File: Library/Formula/muttvcardsearch.rb (hunk 6 above)
Comment (mikemcquaid) [inline Library/Formula/muttvcardsearch.rb]: We should fix up that `make install` instead and submit a patch upstream.

Comment (benjaminfrank) [inline Library/Formula/muttvcardsearch.rb]: Fixed & reported upstream (https://github.com/BlackIkeEagle/muttvcardsearch/pull/10).
In the meantime the formula patches the cmake file .

Comment (mikemcquaid) [inline Library/Formula/muttvcardsearch.rb]: Please add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Also, use `https://github.com/BlackIkeEagle/muttvcardsearch/pull/10.patch` as the patch URL. Thanks!

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: Library/Formula/muttvcardsearch.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  patch do
+    url "https://patch-diff.githubusercontent.com/raw/BlackIkeEagle/muttvcardsearch/pull/10.patch"
+    sha256 "683528161e1b4f19ba75c5e367917969f2d5f995fc753659d8c59079a314d3cc"
+  end
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+      bin.install "muttvcardsearch"
```
Comment (DomT4) [inline Library/Formula/muttvcardsearch.rb]: Is the CMakeList not installing this, even with your patch?

Comment (benjaminfrank) [inline Library/Formula/muttvcardsearch.rb]: Sorry, my bad. Fixed it.

**Comment by benjaminfrank** [comment]
Upstream seems pretty much inactive to incorporate the cmake fix. But then I am no longer using this software anyway.

**Comment by DomT4** [comment]
Alright, let's pass on this if upstream haven't responded to the patch PR and the software is no longer being used by you. Thanks for the submission though! :heart:

**Pull request closed without merging**